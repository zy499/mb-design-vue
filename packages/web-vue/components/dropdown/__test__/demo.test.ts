import demoTest from '../../../scripts/demo-test';

demoTest('dropdown');
