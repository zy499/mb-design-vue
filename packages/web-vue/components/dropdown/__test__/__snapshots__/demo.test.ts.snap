// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<dropdown> demo: render [basic] correctly 1`] = `
"<div class=\\"mb-space mb-space-horizontal mb-space-align-center\\">
  <div class=\\"mb-space-item\\" style=\\"margin-right: 24px;\\"><button class=\\"mb-btn mb-btn-secondary mb-btn-shape-square mb-btn-size-medium mb-btn-status-normal\\" type=\\"button\\">
      <!--v-if-->Click Me
    </button>
    <!---->
  </div>
  <div class=\\"mb-space-item\\" style=\\"margin-right: 24px;\\"><button class=\\"mb-btn mb-btn-secondary mb-btn-shape-square mb-btn-size-medium mb-btn-status-normal mb-btn-disabled\\" type=\\"button\\" disabled=\\"\\">
      <!--v-if-->Click Me
    </button>
    <!---->
  </div>
  <div class=\\"mb-space-item\\"><button class=\\"mb-btn mb-btn-secondary mb-btn-shape-square mb-btn-size-medium mb-btn-status-normal\\" type=\\"button\\">
      <!--v-if-->No Max Height <svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-down\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\">
        <path d=\\"M39.6 17.443 24.043 33 8.487 17.443\\"></path>
      </svg>
    </button>
    <!---->
  </div>
</div>"
`;

exports[`<dropdown> demo: render [button] correctly 1`] = `
"<div class=\\"mb-space mb-space-horizontal mb-space-align-center\\">
  <div class=\\"mb-space-item\\" style=\\"margin-right: 24px;\\">
    <div class=\\"mb-btn-group\\"><button class=\\"mb-btn mb-btn-secondary mb-btn-shape-square mb-btn-size-medium mb-btn-status-normal\\" type=\\"button\\">
        <!--v-if--> Publish
      </button><button class=\\"mb-btn mb-btn-secondary mb-btn-shape-square mb-btn-size-medium mb-btn-status-normal mb-btn-only-icon\\" type=\\"button\\"><span class=\\"mb-btn-icon\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-more\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M38 25v-2h2v2h-2ZM23 25v-2h2v2h-2ZM8 25v-2h2v2H8Z\\" fill=\\"currentColor\\" stroke=\\"none\\"></path><path d=\\"M38 25v-2h2v2h-2ZM23 25v-2h2v2h-2ZM8 25v-2h2v2H8Z\\"></path></svg></span></button>
      <!---->
    </div>
  </div>
  <div class=\\"mb-space-item\\" style=\\"margin-right: 24px;\\">
    <div class=\\"mb-btn-group\\"><button class=\\"mb-btn mb-btn-secondary mb-btn-shape-square mb-btn-size-medium mb-btn-status-normal mb-btn-disabled\\" type=\\"button\\" disabled=\\"\\">
        <!--v-if--> Disabled
      </button><button class=\\"mb-btn mb-btn-secondary mb-btn-shape-square mb-btn-size-medium mb-btn-status-normal mb-btn-disabled mb-btn-only-icon\\" type=\\"button\\" disabled=\\"\\"><span class=\\"mb-btn-icon\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-more\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M38 25v-2h2v2h-2ZM23 25v-2h2v2h-2ZM8 25v-2h2v2H8Z\\" fill=\\"currentColor\\" stroke=\\"none\\"></path><path d=\\"M38 25v-2h2v2h-2ZM23 25v-2h2v2h-2ZM8 25v-2h2v2H8Z\\"></path></svg></span></button>
      <!---->
    </div>
  </div>
  <div class=\\"mb-space-item\\">
    <div class=\\"mb-btn-group\\"><button class=\\"mb-btn mb-btn-secondary mb-btn-shape-square mb-btn-size-medium mb-btn-status-normal\\" type=\\"button\\">
        <!--v-if--> Publish
      </button><button class=\\"mb-btn mb-btn-secondary mb-btn-shape-square mb-btn-size-medium mb-btn-status-normal mb-btn-only-icon\\" type=\\"button\\"><span class=\\"mb-btn-icon\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-down\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M39.6 17.443 24.043 33 8.487 17.443\\"></path></svg></span></button>
      <!---->
    </div>
  </div>
</div>"
`;

exports[`<dropdown> demo: render [context-menu] correctly 1`] = `
"<div style=\\"display: flex; align-items: center; justify-content: center; height: 300px;\\">
  <div>Click Me</div>
</div>
<!---->"
`;

exports[`<dropdown> demo: render [group] correctly 1`] = `
"<button class=\\"mb-btn mb-btn-secondary mb-btn-shape-square mb-btn-size-medium mb-btn-status-normal\\" type=\\"button\\">
  <!--v-if-->Click Me
</button>
<!---->"
`;

exports[`<dropdown> demo: render [icon] correctly 1`] = `
"<button class=\\"mb-btn mb-btn-secondary mb-btn-shape-square mb-btn-size-medium mb-btn-status-normal\\" type=\\"button\\">
  <!--v-if-->Click Me
</button>
<!---->"
`;

exports[`<dropdown> demo: render [position] correctly 1`] = `
"<div class=\\"mb-space mb-space-horizontal mb-space-align-center\\">
  <div class=\\"mb-space-item\\" style=\\"margin-right: 8px;\\"><button class=\\"mb-btn mb-btn-secondary mb-btn-shape-square mb-btn-size-medium mb-btn-status-normal\\" type=\\"button\\">
      <!--v-if-->Bottom Left
    </button>
    <!---->
  </div>
  <div class=\\"mb-space-item\\" style=\\"margin-right: 8px;\\"><button class=\\"mb-btn mb-btn-secondary mb-btn-shape-square mb-btn-size-medium mb-btn-status-normal\\" type=\\"button\\">
      <!--v-if-->Bottom
    </button>
    <!---->
  </div>
  <div class=\\"mb-space-item\\" style=\\"margin-right: 8px;\\"><button class=\\"mb-btn mb-btn-secondary mb-btn-shape-square mb-btn-size-medium mb-btn-status-normal\\" type=\\"button\\">
      <!--v-if-->Bottom Right
    </button>
    <!---->
  </div>
  <div class=\\"mb-space-item\\" style=\\"margin-right: 8px;\\"><button class=\\"mb-btn mb-btn-secondary mb-btn-shape-square mb-btn-size-medium mb-btn-status-normal\\" type=\\"button\\">
      <!--v-if-->Top Left
    </button>
    <!---->
  </div>
  <div class=\\"mb-space-item\\" style=\\"margin-right: 8px;\\"><button class=\\"mb-btn mb-btn-secondary mb-btn-shape-square mb-btn-size-medium mb-btn-status-normal\\" type=\\"button\\">
      <!--v-if-->Top
    </button>
    <!---->
  </div>
  <div class=\\"mb-space-item\\"><button class=\\"mb-btn mb-btn-secondary mb-btn-shape-square mb-btn-size-medium mb-btn-status-normal\\" type=\\"button\\">
      <!--v-if-->Top Right
    </button>
    <!---->
  </div>
</div>"
`;

exports[`<dropdown> demo: render [submenu] correctly 1`] = `
"<button class=\\"mb-btn mb-btn-secondary mb-btn-shape-square mb-btn-size-medium mb-btn-status-normal\\" type=\\"button\\">
  <!--v-if-->Click Me
</button>
<!---->"
`;

exports[`<dropdown> demo: render [trigger] correctly 1`] = `
"<div class=\\"mb-space mb-space-horizontal mb-space-align-center\\">
  <div class=\\"mb-space-item\\" style=\\"margin-right: 24px;\\"><button class=\\"mb-btn mb-btn-secondary mb-btn-shape-square mb-btn-size-medium mb-btn-status-normal\\" type=\\"button\\">
      <!--v-if-->Click Me
    </button>
    <!---->
  </div>
  <div class=\\"mb-space-item\\"><button class=\\"mb-btn mb-btn-secondary mb-btn-shape-square mb-btn-size-medium mb-btn-status-normal\\" type=\\"button\\">
      <!--v-if-->Hover Me
    </button>
    <!---->
  </div>
</div>"
`;
